### PR TITLE
[SYCL][New PM] Move code in sycl-post-link to new pass manager

### PR DESCRIPTION
**Pull request by asudarsa**
In the sycl-post-link tool, most of the changes towards moving the passes to use the new pass manager were added as part of this PR by Konstantin Bobrovsky: https://github.com/intel/llvm/pull/6835
There was one site where legacy pass manager was used. This change moves that to use the new pass manager.

It is sufficient to check for regression in existing tests. There is no need to add a new test.

Thanks
Sincerely

### CODE REVIEW

File: llvm/tools/sycl-post-link/sycl-post-link.cpp [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include "llvm/IR/LLVMContext.h"
 #include "llvm/IR/LegacyPassManager.h"
```
Comment (bader) [inline llvm/tools/sycl-post-link/sycl-post-link.cpp:33]: Do we still need this include?

Comment (asudarsa) [reply llvm/tools/sycl-post-link/sycl-post-link.cpp:33]: Ah..good catch. I do not think we need this include anymore. Thanks. I will remove it in the upcoming commit.

**Comment by asudarsa** [comment]
For test fails, it seems like most of the failures are due to NOT registering MAM. I am adding that change now and testing locally. Thanks

**Review by AlexeySachkov — APPROVED** [review]

**Review by v-klochkov — APPROVED** [review]
Please make the title a bit more specific by mentioning sycl-post-link.
This PR moves only 1 file to new pass manager

**Comment by asudarsa** [comment]
Hi @bader and @AlexeySachkov 

Can one of you please help with merging this?

Thanks
Sincerely

**Comment by dm-vodopyanov** [comment]
> Hi @bader and @AlexeySachkov
> 
> Can one of you please help with merging this?
> 
> Thanks Sincerely

https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md#merge

**Comment by aelovikov-intel** [comment]
This broke post-commit: https://github.com/intel/llvm/actions/runs/5612957520/job/15208195304. Please fix ASAP.

**Comment by asudarsa** [comment]
Looking at this now. Thanks

**Pull request merged**